addons/disable-auto-save/userscript.js

```diff
@@ -1,13 +1,79 @@
 export default async ({ addon, console, msg }) => {
+  function appendChildWithAttributes(parent, type, attrs = {}) {
+    const element = document.createElement(type);
+    Object.assign(element, attrs);
+    parent.appendChild(element);
+    return element;
+  }
+  function setProjectChanged(changed) {
+    addon.tab.redux.dispatch({
+      type: "scratch-gui/project-changed/SET_PROJECT_CHANGED",
+      changed,
+    });
+  }
+  const PAGE_BUTTON_SELECTOR = "[class*='community-button_community-button_']";
+
   addon.tab.redux.initialize();
   addon.tab.redux.addEventListener("statechanged", ({ detail }) => {
     if (addon.self.disabled || detail.action.type !== "timeout/SET_AUTOSAVE_TIMEOUT_ID") return;
     clearTimeout(detail.next.scratchGui.timeout.autoSaveTimeoutId);
     console.log("Pending autosave prevented.");
   });
 
+  function showSaveAlert() {
```

Comment:
This modal implementation looks great and works well! However, as you credited it shares many similarities with `asset-conflict-dialog` addon's modal.

 Since we now have two addons using this same modal pattern (createModal + button row + scratch
  classes), would it make sense to abstract this into a shared utility or add it to the `addon.tab`       
  API? This would ensure code remains DRY, is consistent and makes future similar modals easier.
  
   Something like:
   ```js
  addon.tab.createPromptModal(title, {
    subtitle: "...",
    message: "...",
    buttons: [
      { value: "cancel", label: msg("cancel") },
      { value: "save", label: msg("save"), primary: true }
    ],
    onButtonClick: (value) => { /* handle */ }
  });
  ```

Reply:
I personally would do it in the same PR, otherwise you're making technical debt that is to be resolved later. @Samq64 What do you think?

Reply:
I think either way is fine. `animated-thumb`, `studio-followers` and `mediarecorder` could also work with your example. `asset-conflict-dialog` uses a different class for non-primary buttons compared to other addons though.

`blocks2image` has button icons and one that is left-aligned. Does it make sense to expand the API to allow that?